test/plugin_functional/test_suites/shared_ux/eui_provider.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import expect from '@kbn/expect';
+import { PluginFunctionalProviderContext } from '../../services';
+
+export default function ({ getPageObjects, getService }: PluginFunctionalProviderContext) {
+  const lifecycle = getService('lifecycle');
+  const PageObjects = getPageObjects(['common', 'header']);
+  const testSubjects = getService('testSubjects');
+  const browser = getService('browser');
+  const log = getService('log');
+
+  lifecycle.afterTestSuite.add(async () => {
+    // check sessionStorage for trace of EuiProvider dev error thrown
+    const [errorMessage, errorStack, pageHref, pageTitle] = await Promise.all([
+      browser.getSessionStorageItem('dev.euiProviderWarning.message'),
+      browser.getSessionStorageItem('dev.euiProviderWarning.stack'),
+      browser.getSessionStorageItem('dev.euiProviderWarning.pageHref'),
+      browser.getSessionStorageItem('dev.euiProviderWarning.pageTitle'),
+    ]);
+    if (errorMessage != null) {
+      log.error(`pageTitle: ${pageTitle}`);
+      log.error(`pageHref: ${pageHref}`);
+      log.error(`Error: ${errorMessage}`);
+      log.error(`Error stack: ${errorStack}`);
+      throw new Error(`Found EuiProvider dev error on: ${pageHref}`);
+    }
+  });
```

Comment:
@elastic/appex-qa I need some guidance on where this block of code should be placed in order to catch implementation errors with EuiProvider that could happen in any application.

My feeling is that it doesn't belong here, but probably some service such as https://github.com/elastic/kibana/blob/main/packages/kbn-ftr-common-functional-ui-services/services/remote/remote.ts#L94